src/ttt_core/console.clj

```diff
@@ -55,19 +60,45 @@
   (do
     (clear-terminal)
     (draw-header)
-    (draw-player-info (:player-1-token game) (:player-2-token game))
+    (draw-player-info game)
     (draw-board (:board game))
     (println message)
     (draw-footer)))
 
-(defn build-current-player-string [game] 
-  (if (= :player-1-token (:current-token game))
-    "Player 1's move!"
-    "Player 2's move!"))
+(defn build-choose-game-mode-string 
+  ([] (build-choose-game-mode-string ""))
+  ([message]
+    (str message "Choose a game mode:\n1. Human-vs-Human\n2. Human-vs-Computer")))
+
+(defn request-game-mode
+  ([game] (request-game-mode game ""))
+  ([game message]
+    (let [modes {:1 :human-vs-human :2 :human-vs-computer}]
+      (do
+        (draw-main game "--- Game Mode Selection ---")
+        (println (build-choose-game-mode-string message))
+        (get modes (keyword (read-line)))))))
+
+(defn get-game-mode-selection [game] 
+  (loop [selection (request-game-mode game)]
+    (if (not (nil? selection))
+      selection
+      (recur (request-game-mode game "Invalid selection. ")))))
+
+(defn handle-game-mode-selection [game]
+  (assoc game :game-mode (get-game-mode-selection game)))
+
+(defn build-current-player-string [game]
```

Comment:
This could be refactored to pass just what the function needs. That way instead of the console needing to dig into the game to check whos turn it is and what game mode it is (sounds like a lot for the console to know) you could just hand it the current player and the game mode.

Reply:
Addressed in new commit by updating `build-current-player-string` to accept only a player name. A separate function, `current-player-name` accepts the current token and game mode to identify the name of the current player. This function is used to get that name and then pass the value into `build-current-player-string`. I considered putting the `current-player-name` function in the `ttt-core.game` namespace, but doing so would have caused a circular dependency. Much of the console already has similar functions dealing with a game object so I thought this was a reasonable solution.